src/EnrollDuoController.js

```diff
@@ -11,118 +11,113 @@
  */
 
 /* eslint camelcase: 0*/
-define([
-  'okta',
-  'duo',
-  'q',
-  'util/FormController',
-  'views/enroll-factors/Footer'
-],
-function (Okta, Duo, Q, FormController, Footer) {
+import { $, _, loc } from 'okta';
+import Duo from 'duo';
+import Q from 'q';
+import FormController from 'util/FormController';
+import Footer from 'views/enroll-factors/Footer';
+export default FormController.extend({
+  className: 'enroll-duo duo-form',
 
-  var $ = Okta.$,
-      _ = Okta._;
+  Model: {
+    props: {
+      host: 'string',
+      signature: 'string',
+      postAction: 'string',
+      factorId: 'string',
+      stateToken: 'string',
+    },
 
-  return FormController.extend({
+    getInitOptions: function () {
+      return this.doTransaction(function (transaction) {
+        const factor = _.findWhere(transaction.factors, {
+          factorType: 'web',
+          provider: 'DUO',
+        });
 
-    className: 'enroll-duo duo-form',
+        return factor.enroll();
+      });
+    },
 
-    Model: {
-      props: {
-        host: 'string',
-        signature: 'string',
-        postAction: 'string',
-        factorId: 'string',
-        stateToken: 'string'
-      },
+    activate: function (signedResponse) {
+      const url = this.get('postAction');
+      const factorId = this.get('factorId');
+      const self = this;
+      const data = {
+        id: factorId,
+        stateToken: this.get('stateToken'),
+        sig_response: signedResponse,
+      };
+      // Note: We should be doing this in OktaAuth! Fix when it's updated.
```

Comment:
Is there a related JIRA for this?

Reply:
I don think so. I wasn't even aware why it says shall be done in OktaAuth.

Reply:
OK, I just have some confusion on what the actual issue is here. Therefore I have no confidence that any "fix" will be put in.